src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBadge/InfoBadge.cs

```diff
@@ -1,108 +1,109 @@
+			if (Value < -1)
 			{
-				return new Size(defaultDesiredSize.Height, defaultDesiredSize.Height);
+				throw new ArgumentOutOfRangeException("Value must be equal to or greater than -1");
 			}
-			return defaultDesiredSize;
 		}
 
-		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		if (property == ValueProperty || property == IconSourceProperty)
 		{
-			var property = args.Property;
-			Control thisAsControl = this;
-
-			if (property == ValueProperty)
-			{
-				if (Value < -1)
-				{
-					throw new ArgumentOutOfRangeException("Value must be equal to or greater than -1");
-				}
-			}
-
-			if (property == ValueProperty || property == IconSourceProperty)
-			{
-				OnDisplayKindPropertiesChanged();
-			}
+			OnDisplayKindPropertiesChanged();
 		}
+	}
 
-		void OnDisplayKindPropertiesChanged()
+	void OnDisplayKindPropertiesChanged()
+	{
+		Control thisAsControl = this;
+		if (Value >= 0)
 		{
-			Control thisAsControl = this;
-			if (Value >= 0)
-			{
-				VisualStateManager.GoToState(thisAsControl, "Value", true);
-			}
-			else if (IconSource is { } iconSource)
+			VisualStateManager.GoToState(thisAsControl, "Value", true);
+		}
+		else if (IconSource is { } iconSource)
+		{
+			TemplateSettings.IconElement = iconSource.CreateIconElement();
+			if (iconSource is FontIconSource)
 			{
-				TemplateSettings.IconElement = iconSource.CreateIconElement();
-				if (iconSource is FontIconSource)
-				{
-					VisualStateManager.GoToState(thisAsControl, "FontIcon", true);
-				}
-				else
-				{
-					VisualStateManager.GoToState(thisAsControl, "Icon", true);
-				}
+				VisualStateManager.GoToState(thisAsControl, "FontIcon", true);
 			}
 			else
 			{
-				VisualStateManager.GoToState(thisAsControl, "Dot", true);
+				VisualStateManager.GoToState(thisAsControl, "Icon", true);
 			}
 		}
+		else
+		{
+			VisualStateManager.GoToState(thisAsControl, "Dot", true);
+		}
+	}
 
-		private void OnSizeChanged(object sender, Windows.UI.Xaml.SizeChangedEventArgs args)
+	private void OnSizeChanged(object sender, Windows.UI.Xaml.SizeChangedEventArgs args)
+	{
+		CornerRadius GetCornerRadius()
 		{
-			CornerRadius GetCornerRadius()
+			var cornerRadiusValue = ActualHeight / 2;
+			if (ReadLocalValue(CornerRadiusProperty) == DependencyProperty.UnsetValue)
 			{
-				var cornerRadiusValue = ActualHeight / 2;
-				if (SharedHelpers.IsRS5OrHigher())
-				{
-					if (ReadLocalValue(CornerRadiusProperty) == DependencyProperty.UnsetValue)
-					{
-						return new CornerRadius(cornerRadiusValue, cornerRadiusValue, cornerRadiusValue, cornerRadiusValue);
-					}
-					else
-					{
-						return new CornerRadius();
-					}
-				}
 				return new CornerRadius(cornerRadiusValue, cornerRadiusValue, cornerRadiusValue, cornerRadiusValue);
 			}
+			else
+			{
+				return new CornerRadius();
+			}
+			//return new CornerRadius(cornerRadiusValue, cornerRadiusValue, cornerRadiusValue, cornerRadiusValue);
```

Comment:
Is it commented out in the original version?

Reply:
No, but the code is unreachable, they just didn't remove it, so I kept it commented out.